### DISCUSSION TITLE
publish values via MQTT

### DISCUSSION THREAD

**Discussion by MarkusH1975** [michaelarnauts/aiocomfoconnect#1] [Ideas]
Hi,  

here is a quick hack to publish values on MQTT server.

python -u -m aiocomfoconnect show-sensors --uuid "00000000000000000000000000000002" --host 192.168.x.x | sed -Eu "s/\s*(.*):\s+(.*)\s+(.*)/\/aiocomfoconnect\/\1: \2/g" | sed -Eu "s/[ ()]/_/g" | sed -Eu "s/(.*):_(.*)/mosquitto_pub -h localhost -t \1 -m \2/g" | /bin/bash

It would be really nice to have similar function within your python script and run it as daemon.

In this way it can be easily used by different home automation systems.

Regards,
Markus

**Comment by michaelarnauts** [comment]
Cool idea, but this is out of scope for this library, although it shouldn't be that hard to create a new python project that uses this library and pushes the sensor values to mqtt (and even listen for commands to change the fan speed for example).

If I remember correctly, there are some projects that use the old comfoconnect library and do just this.

**Comment by oysteing** [comment]
Hi, I've just created a project to publish values to an MQTT server. As of the first version, it can only publish all sensor values. Take a look if you're interested. I'm curious to hear about your experiences.

**Comment by michaelarnauts** [comment]
Awesome! I'll gladly add a link to the README to your project once you feel it's ready.

I do remember seeing an issue somewhere about a request for this. I'll see if I can find it.

Link to repository: https://github.com/oysteing/comfoconnect-mqtt-bridge

**Comment by oysteing** [comment]
Cool, I've added the ability to publish selected sensor values. It's been running without issues for a month now. I've created a 1.0.0 release and would be happy to be referenced from the README.

Cheers,
Øystein
